apps/meteor/app/api/server/v1/channels.ts

```diff
@@ -565,22 +565,36 @@ API.v1.addRoute(
 	{ authRequired: true },
 	{
 		async post() {
-			const findResult = await findChannelByIdOrName({
-				params: this.bodyParams,
-				checkedArchived: false,
-			});
+			let roomId = 'roomId' in this.bodyParams ? this.bodyParams.roomId : undefined;
+			let roomName = 'roomName' in this.bodyParams ? this.bodyParams.roomName : undefined;
 
-			const sub = await Subscriptions.findOneByRoomIdAndUserId(findResult._id, this.userId);
+			try {
+				const findResult = await findChannelByIdOrName({
+					params: this.bodyParams,
+					checkedArchived: false,
+				});
+				roomName = findResult.name;
+
+				if (findResult._id) {
+					roomId = findResult._id;
+				}
+			} catch (e) {}
+
+			if (!roomId) {
+				return API.v1.failure('Could not find the channel or any subscription linked to it');
+			}
```

Comment:
Please use `validateParams` in the endpoint definition instead to make sure all params are correctly sent.